orb-jobs-agent/src/handlers/beacon.rs

```
@@ -0,0 +1,37 @@
+use crate::job_system::ctx::{Ctx, JobExecutionUpdateExt};
+use color_eyre::{eyre::WrapErr, Result};
+use orb_relay_messages::jobs::v1::JobExecutionUpdate;
+
+/// command format: `beacon`
+#[tracing::instrument(skip(ctx))]
+pub async fn handler(ctx: Ctx) -> Result<JobExecutionUpdate> {
+    let output = ctx
+        .deps()
+        .shell
+        .exec(&["orb-beacon"])
+        .await
+        .wrap_err("failed to spawn orb-beacon")?
+        .wait_with_output()
+        .await
+        .wrap_err("failed to get orb-beacon output")?;
+
+    if !output.status.success() {
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        return Ok(ctx.failure().stderr(stderr));
```

you can also just return an error, will have the same effect and be a bit more idiomatic 

after factor, all failures are gracefully handled and sent back as an error to the backend